Neos.Flow/Classes/Cache/CacheManager.php

```diff
@@ -402,6 +414,42 @@ protected function flushConfigurationCachesByChangedFiles(array $changedFiles):
         }
     }
 
+    /**
+     * A slot that flushes caches as needed if classes with specific annotations have changed
+     *
+     * @param array<string> $classNames The full class names of the classes that got compiled
+     * @return void
+     */
+    public function flushConfigurationCachesByCompiledClass(array $classNames): void
+    {
+        $caches = [
+            Neos\Flow\Annotations\Route::class => ['Flow_Mvc_Routing_Route', 'Flow_Mvc_Routing_Resolve']
+        ];
+        $cachesToFlush = [];
+
+        foreach ($classNames as $className) {
+            foreach ($caches as $annotationClass => $cacheNames) {
+                if (!$this->reflectionService->isClassAnnotatedWith($className, $annotationClass)
+                    && count($this->reflectionService->getMethodsAnnotatedWith($className, $annotationClass)) === 0) {
+                    continue;
+                }
+                foreach ($caches[$annotationClass] as $cacheName) {
+                    $cachesToFlush[$cacheName] = $annotationClass;
+                }
+            }
+        }
+
+        foreach ($cachesToFlush as $cacheName => $annotationClass) {
+            $this->logger->info(sprintf('A class file containing the annotation "%s" has been changed, flushing related cache "%s"', $annotationClass, $cacheName), LogEnvironment::fromMethodName(__METHOD__));
+            $this->getCache($cacheName)->flush();
+        }
+
+        if (count($cachesToFlush) > 0) {
```

Comment:
Not sure, but I think this `count()` check should also happen in the other `flush*Caches...()` methods, no? But maybe I'm missing something